--- src/main/kotlin/coffee/cypher/hexbound/feature/construct/entity/SpiderConstructEntity.kt
+++ src/main/kotlin/coffee/cypher/hexbound/feature/construct/entity/SpiderConstructEntity.kt
@@ -131,8 +131,8 @@
         return redirectSpiderLang(super.getDefaultName(), this)
     }

-    override fun registerControllers(controllers: AnimatableManager.ControllerRegistrar) {
-        controllers.add(animationController)
+    override fun registerControllers(data: software.bernie.geckolib.animatable.instance.AnimatableInstanceCache?) {
+
     }

     override fun getAnimatableInstanceCache(): AnimatableInstanceCache {
