--- src/main/kotlin/coffee/cypher/hexbound/feature/construct/entity/SpiderConstructEntity.kt
+++ src/main/kotlin/coffee/cypher/hexbound/feature/construct/entity/SpiderConstructEntity.kt
@@ -165,11 +165,11 @@
         private val DANCE_ANIMATION = RawAnimation.begin().thenLoop("dance")

         val ALT_MODEL_ENABLED: EntityDataAccessor<Boolean> = SynchedEntityData.defineId(
-            SpiderConstructEntity::class.java as Class<SpiderConstructEntity>,
+            SpiderConstructEntity::class.java,
             EntityDataSerializers.BOOLEAN
         )

         val HELD_STACK: EntityDataAccessor<ItemStack> = SynchedEntityData.defineId(
-            SpiderConstructEntity::class.java as Class<SpiderConstructEntity>,
+            SpiderConstructEntity::class.java,
             EntityDataSerializers.ITEM_STACK
         )
