--- src/main/kotlin/coffee/cypher/hexbound/feature/media_attachment/StaticMediaValue.kt
+++ src/main/kotlin/coffee/cypher/hexbound/feature/media_attachment/StaticMediaValue.kt
@@ -21,8 +21,8 @@
     val value get() = unit.calculate(amount)

     companion object {
-        val UNIT_CODEC: Codec<Unit> = StringRepresentable.fromEnum({ Unit.values() })
-        val UNIT_STREAM_CODEC: StreamCodec<ByteBuf, Unit> = ByteBufCodecs.idMapper({ it.ordinal }, { i -> Unit.values()[i] })
+        val UNIT_CODEC: Codec<Unit> = StringRepresentable.fromEnum({ Unit.entries.toTypedArray() })
+        val UNIT_STREAM_CODEC: StreamCodec<ByteBuf, Unit> = ByteBufCodecs.idMapper({ it.ordinal }, { i -> Unit.entries.toTypedArray()[i] })

         val CODEC: Codec<StaticMediaValue> = RecordCodecBuilder.create { instance ->
             instance.group(
